flow/src/main/java/com/sequenceiq/flow/reactor/api/event/BaseFlowEvent.java

```diff
@@ -1,24 +1,28 @@
 package com.sequenceiq.flow.reactor.api.event;
 
 import com.sequenceiq.cloudbreak.common.event.Acceptable;
+import com.sequenceiq.cloudbreak.common.event.ResourceCrnPayload;
 import com.sequenceiq.cloudbreak.common.event.Selectable;
 
 import reactor.rx.Promise;
 
-public class BaseFlowEvent implements Selectable, Acceptable {
+public class BaseFlowEvent implements Selectable, Acceptable, ResourceCrnPayload {
```

Comment:
Events should not implement the Payload interface. Payloads are used in the generic parameters reactor.bus.Event<> class. Instead rename `ResourceCrnPayload` to `ResourceAwareFlowEvent` and rename `BaseNamedFlowEvent` to `BaseResourceFlowEvent`

Reply:
Selectable and Acceptable implements Payload as well.

Reply:
indeed

Reply:
The env actions use the Payload interface like `AbstractVpcDeleteAction` or `AbstractVpcCreateAction`